R/collapse.R

```diff
@@ -23,10 +23,14 @@ fct_collapse <- function(.f, ..., group_other = FALSE) {
     f <- check_factor(.f)
     levels <- levels(f)
     new[["Other"]] <- levels[!levels %in% levs]
-    levs <- levels
+    levs <- as.list(unlist(new, use.names = FALSE))
   }
 
   names(levs) <- names(new)[rep(seq_along(new), vapply(new, length, integer(1)))]
 
-  fct_recode(.f, !!!levs)
+  out <- fct_recode(.f, !!!levs)
+
+  if (any(levels(out) == "Other")){
+    fct_relevel(out, "Other", after = Inf)
```

Comment:
Could we do this move earlier? (i.e. by adjusting `levs`?)

Reply:
Without doing a deeper dive, I don't _think_ so.  The issue is that `check_factor` applies the default factor order of `base::factor`, meaning it will be a lottery as to whether 'Other' will appear last.

If it's the `if` operator which is bothersome, perhaps I could just delete the entire _if/else_ and manually provide the levels in the right order:
```
    levels(out) <- names(new)
```
If there is an _Other_, then it will already be last in `names(new)`. If not, nothing will change.

Reply:
In short, I don't think so. I have reworked a little though, so it is consistent with `fct_other()`

Reply:
Ok, in that case, can you please fix the style: https://style.tidyverse.org/syntax.html#indenting